splight_lib/client/datalake/remote_client.py

```diff
@@ -277,22 +277,111 @@ def _flusher(self):
+        url = self._base_url / f"{self._PREFIX}/write"
+        collection = camelcase(collection)
+        data = {
+            "collection": collection,
+            "records": docs,
+        }
+        response = self._restclient.post(url, json=data)
+        response.raise_for_status()
+        return docs
+
+
+class BufferedSyncRemoteDataClient(SyncRemoteDatalakeClient):
+    _PREFIX = "data"
+
+    def __init__(
+        self,
+        base_url: str,
+        access_id: str,
+        secret_key: str,
+        buffer_size: int = DL_BUFFER_SIZE,
+        buffer_timeout: float = DL_BUFFER_TIMEOUT,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(
+            base_url,
+            access_id,
+            secret_key,
+            buffer_size,
+            buffer_timeout,
+            *args,
+            **kwargs,
+        )
+        self._base_url = furl(base_url)
+        token = SplightAuthToken(
+            access_key=access_id,
+            secret_key=secret_key,
+        )
+        self._restclient = SplightRestClient()
+        self._restclient.update_headers(token.header)
+
+        logger.debug(
+            "Initializing buffer with size %s and timeout %s",
+            buffer_size,
+            buffer_timeout,
+        )
+        self._data_buffers = {
+            "default": DatalakeDocumentBuffer(buffer_size, buffer_timeout),
+            "routine_evaluations": DatalakeDocumentBuffer(
+                buffer_size, buffer_timeout
+            ),
+        }
+        self._lock = Lock()
+        logger.debug(
+            "Synchronous Buffered Remote datalake client initialized.",
+            tags=LogTags.DATALAKE,
+        )
+
+    def save(
+        self, collection: str, instances: Union[List[Dict], Dict]
+    ) -> List[Dict]:
+        logger.debug("Saving documents in datalake")
+        instances = instances if isinstance(instances, List) else [instances]
+        if collection not in self._data_buffers:
+            raise InvalidCollectionName(collection)
+        buffer = self._data_buffers[collection]
+        with self._lock:
+            if buffer.should_flush():
                 logger.debug(
                     "Flushing datalake buffer with %s elements",
                     len(buffer.data),
                 )
                 self._send_documents(collection, buffer.data)
                 buffer.reset()
-            except Exception:
-                logger.error("Unable to save documents", exc_info=True)
+            buffer.add_documents(instances)
+        return instances
+
+    def save_dataframe(
+        self, dataframe: pd.DataFrame, collection: str = DEFAULT_COLLECTION
+    ) -> None:
+        logger.debug("Saving dataframe in datalake")
+        dataframe["timestamp"] = dataframe["timestamp"].apply(
+            lambda x: x.tz_localize(tz="UTC").isoformat()
+        )
+        instances = dataframe.to_dict("records")
```

Comment:
Have you checked what happens when the "value" column has number and string at the same time?